src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java

```diff
@@ -859,24 +882,31 @@ public void onDeath(final DamageSource par1DamageSource)
         if (colony != null)
         {
             colony.decreaseOverallHappiness(penalty);
-            triggerDeathAchievement(par1DamageSource, getColonyJob());
+            triggerDeathAchievement(damageSource, getColonyJob());
             if (getColonyJob() instanceof JobGuard)
             {
                 LanguageHandler.sendPlayersMessage(
                         colony.getMessageEntityPlayers(),
                         "tile.blockHutTownHall.messageGuardDead",
-                        citizenData.getName(), (int) posX, (int) posY, (int) posZ);
+                        citizenData.getName(), (int) posX, (int) posY, (int) posZ, damageSource.damageType);
             }
             else
             {
                 LanguageHandler.sendPlayersMessage(
                         colony.getMessageEntityPlayers(),
                         "tile.blockHutTownHall.messageColonistDead",
-                        citizenData.getName(), (int) posX, (int) posY, (int) posZ);
+                        citizenData.getName(), (int) posX, (int) posY, (int) posZ, damageSource.damageType);
             }
             colony.removeCitizen(getCitizenData());
         }
-        super.onDeath(par1DamageSource);
+        super.onDeath(damageSource);
+    }
+
+    @Override
+    public EnumActionResult applyPlayerInteraction(final EntityPlayer player, final Vec3d vec, final EnumHand hand)
+    {
+        SoundUtils.playInteractionSoundAtCitizenWithChance(CompatibilityUtils.getWorld(this), this.getPosition(), 100, this);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Assign this magic number 100 to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AS109)